frontend/test/acceptance/JoinSupporterSpec.scala

```diff
@@ -68,12 +68,33 @@ class JoinSupporterSpec extends FeatureSpec with Browser
       When("Users fill in delivery address details,")
       enterDetails.fillInDeliveryAddress()
 
-      And("fill in card details,")
-      enterDetails.fillInCardDetails()
+      And("click Continue")
+      enterDetails.clickContinue()
 
       And("click 'Pay' button,")
       enterDetails.pay()
 
+      Then("the Stripe Checkout iframe should display")
+      assert(enterDetails.stripeCheckoutHasLoaded())
+
+      When("the checkout iframe is present")
+      enterDetails.switchToStripe()
+
+      Then("credit card field is present")
+      assert(enterDetails.stripeCheckoutHasCC())
+
+      Then("expiry date field is present")
+      assert(enterDetails.stripeCheckoutHasExph())
+
+      Then("CVC field is present")
+      assert(enterDetails.stripeCheckoutHasCVC())
+
+      Then("submit button is present")
+      assert(enterDetails.stripeCheckoutHasSubmit())
```

Comment:
I guess you are asserting on the existence of CC, Exp, CVC and Submit button, so that you can safely go ahead and fill in the form. You should design your test such that `stripeCheckoutHasLoaded()` already gives you this confidence, so you do not need to make multiple extra assertions. If the submit button is there it is very likely strip checkout has loaded.